src/gc/gc.cpp

```diff
@@ -35744,6 +35744,10 @@ void gc_heap::do_post_gc()
                          (uint32_t)settings.reason,
                          !!settings.concurrent);
 
+#if defined(__linux__)
+    GCToEEInterface::UpdateGCEventStatus();
```

Comment:
Updating at the end of the GC means it is impossible to get events during the 1st GC of the process no? I was expecting we'd want this check at the beginning rather than the end.

Reply:
Agreed, it would be good to do the check prior to GC start.  This is also useful when profiling applications where GCs don't occur very often.

Reply:
hmm, I had this feedback but for some reason I guess I failed to enter it in CF... thanks for pointing it out! yeah we do want this at the beginning of a GC.

Reply:
although I should point out that even with this done at the beginning of a GC, it's done *after* suspension already happened so it will not get the SuspendEEStart/End events so this still wouldn't capture the complete sequence of that first GC. if we really want a complete job we should also call it during GC initialization - I would recommend in `gc_heap::initialize_gc`.